### PR TITLE
[assertj] Conditions to test Objects and Lists

### PR DESCRIPTION
**Pull request by stbischof**
AssertJ provides an way to Extend Asserations [see AssertJ](https://assertj.github.io/doc/#assertj-core-conditions)

Conditions helpes to test Objects and Lists using is/has, isNot/hasNot, contains, doesNotContain ...
They are kind of a described Predicate.

This PR provides a these Conditions.

Would be nice to get some FeedbackThe
The output-message of join-Conditions will look better in next assertj version. [fix in assertj](https://github.com/assertj/assertj-core/pull/2074)

```java
       List<ServiceEvents> serviceEvents;
    
        // ListAsserts in combination with Conditions
        serviceEvents
            .areAtLeast(1, typeModified())
            .are(hasObjectClass(A.class))
            .areExactly(3, objectClass(A.class))
            .areNot(typeModifiedEndmatch())

           serviceEvents
            .element(0)
            .is(typeRegistered())
            .is(objectClass(A.class))
            .is(typeRegistered(A.class))
            .is(containServiceProperty(k1, v1))
            .is(matches(REGISTERED, A.class, dictionaryOf(k1, v1)));
```

### CODE REVIEW

**Comment by bjhargrave** [comment]
This is hard to understand how this is supposed to be used. That is, LOTS more javadoc is needed with descriptions and example usage.

**Comment by stbischof** [comment]
> This is hard to understand how this is supposed to be used. That is, LOTS more javadoc is needed with descriptions and example usage.

@bjhargrave Could you have a look on the doc of the Conditions of BundleEvent https://github.com/osgi/osgi-test/pull/256/commits/1f90fd3fba2bb4a137b56661a0d8597cdf9f7a15 and give me a hit whether this matches what you expect?

**Comment by stbischof** [comment]
> This is hard to understand how this is supposed to be used. That is, LOTS more javadoc is needed with descriptions and example usage.

@bjhargrave Could you have a look on the doc and give me a hint whether this matches what you expect?

**Comment by bjhargrave** [comment]
> The output-message of join-Conditions will look better in next assertj version. [fix in assertj](https://github.com/assertj/assertj-core/pull/2074)

AssertJ 3.19.0 is released and our build is updated to use it now.

**Comment by stbischof** [comment]
Will update it when https://github.com/osgi/osgi-test/pull/253 (ServiceReferenceAssert) is merged this PR needs the Dictionary From Servicereference.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Mapped.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package org.osgi.test.assertj.conditions;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.assertj.core.api.Condition;
+
+public class Mapped<FROM, T> extends Condition<FROM> {
```
Comment (stbischof) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Mapped.java:8]: This class may be replaced/changed after https://github.com/assertj/assertj-core/pull/2103

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Descriptive.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+
+import org.assertj.core.api.Condition;
+
+/**
+ * Descriptive that shows the expected and the tested value in the
+ * description.
+ */
+public final class Descriptive<T, EXPECTED> extends Condition<T> {
```
Comment (stbischof) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Descriptive.java:12]: This Class may changes/be replaced  with 
https://github.com/assertj/assertj-core/pull/2083
https://github.com/assertj/assertj-core/pull/2080

Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Descriptive.java:12]: This Commits supports the Status of an Condition 

https://github.com/assertj/assertj-core/commit/771ed6b685f22f37e1a14f8e5d954fa6e6f03c09

**Comment by stbischof** [comment]
rebase on main and also switched the copyright header to Eclipse.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Mapped.java (hunk 1 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/Mapped.java]: Mapped is now part of assertj-core

https://github.com/assertj/assertj-core/pull/2193

**Comment by stbischof** [comment]
Rebased and used the new Mapped- and Verbose-Condition from AssertJ

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/BitmapUtil.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package org.osgi.test.assertj.conditions;
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/BitmapUtil.java:1]: Missing license notice comment

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionAssert.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package org.osgi.test.assertj.testutil;
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionAssert.java:1]: License comment

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package org.osgi.test.assertj.testutil;
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java:1]: license comment

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/BitmapUtil.java (hunk 3 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/conditions/BitmapUtil.java:19]: added.

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionAssert.java (hunk 4 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionAssert.java:19]: added.

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java (hunk 5 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java:19]: added.

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionAssert.java (hunk 4 above)
Comment (bjhargrave) [reply org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionAssert.java:19]: All the test code should be under `org.osgi.test.assertj.test.*`.

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.assertj.testutil;
+
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java:20]: All the test code should be under `org.osgi.test.assertj.test.*`.

Comment (stbischof) [reply org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/testutil/ConditionMethod.java]: moved into the existing test.testutils package

**Comment by stbischof** [comment]
I would like to add some more conditions in an extra PR
especially for 
- ConfigurationEvents
- OSGiFeatures
- RuntimeDTO's

not all of them are part of the Framework.

So I am not sure what the best place would be.
should all Framework Conditions move from
`org.osgi.test.assertj.conditions` to `org.osgi.test.assertj.framework-conditions`

It might be better so split the Conditions-Class into:

* BundleConditions
* BundleEventConditions
* DictionaryConditions
* FrameworkEventConditions
* ServiceEventConditions
* ServiceReferenceConditions

What do you think @bjhargrave @kriegfrj

**Comment by bjhargrave** [comment]
Do conditions need their own package? 
Can't framework conditions just be placed in the proper existing packages?

**Comment by stbischof** [comment]
@bjhargrave This PR Seems complete to me. Could you please review?

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.condition.VerboseCondition;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.ServiceReference;
+
+/**
+ * A Utility-Class thats Provides static methods to create {@link Condition}s
+ * for an {@link Bundle}
+ */
+public interface BundleConditions {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java:34]: What are these static condition methods in an interface? That seems wrong. I don't see that we want anyone to "implement" these interfaces. Better that these classes are final classes with an empty private constructor.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java [hunk 8]
```diff
@@ -0,0 +1,829 @@
...
+	 * 		.have(typeInstalled()))
+	 * 		.filteredOn(typeInstalled()))
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(typeInstalled());// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<BundleEvent> typeInstalled() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java:316]: I don't think we want to have methods for each event type.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/dictionary/DictionaryConditions.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import org.osgi.framework.Filter;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+/**
+ * A Utility-Class thats Provides static methods to create {@link Condition}s
+ * for an {@link Dictionary}
+ */
+public interface DictionaryConditions {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/dictionary/DictionaryConditions.java:38]: These aren't general dictionary conditions. They are service property conditions. That is, `Dictionary<String,Object>`.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java [hunk 10]
```diff
@@ -0,0 +1,708 @@
...
+	 * 		.filteredOn(typeError())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(typeError());// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<FrameworkEvent> typeError() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java:356]: I don't think we want method for each event type.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java [hunk 11]
```diff
@@ -0,0 +1,605 @@
...
+	 * 		.filteredOn(typeModified())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(typeModified());// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	static Condition<ServiceEvent> typeModified() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java:348]: Same here. Not a method for each event type. It makes the API big and thus make a large testing surface for our tests.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+	 * 				// ObjectAssert}
+	 * 		.has(servicePropertiesIsNotNull());// used on {@link
+	 * 											// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+
+	/**
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java:169]: This appears unfinished. TODOs

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/testutil/ConditionAssert.java [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.regex.Pattern;
+
+import org.assertj.core.api.AbstractThrowableAssert;
+import org.assertj.core.api.Condition;
+import org.assertj.core.api.ObjectAssertFactory;
+
+public interface ConditionAssert {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/testutil/ConditionAssert.java:29]: Isn't this class in another PR?

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java (hunk 12 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java:169]: this is prepared for the moment when we rely on R7. The methods does not exist in R6

File: org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/testutil/ConditionAssert.java (hunk 13 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/test/testutil/ConditionAssert.java:29]: yes. the first merged PR get it in wins. will remove it from the other PRs

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java (hunk 11 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java:348]: reduced to just one `type(int mask)` .

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java (hunk 10 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java:356]: reduced to just one `type(int mask)` .

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java (hunk 8 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java:316]: reduced to just one `type(int mask)` .

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/dictionary/DictionaryConditions.java (hunk 9 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/dictionary/DictionaryConditions.java:38]: renamed to `ServicePropertiesConditions`

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java (hunk 7 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java:34]: yes , changed to final class

**Comment by stbischof** [comment]
@bjhargrave  i did the requested changes

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * public static void example_throwableOfClass(Class<? extends Throwable> throwableClass) {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(throwableOfClass(throwableClass))
+	 * 		.filteredOn(throwableOfClass(throwableClass))
+	 * 		.first()// map to
+	 * 				// {@link
+	 * 				// ObjectAssert}
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java:251]: Some of the formatting in the javadoc comments looks weird. I don't think you want to use `{@link}` in the example code.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.is(sameAs(bundle));// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param bundle - the expected bundle that would be checked against other
+	 *            {@link Bundle}
+	 * @return the Condition<br>
+	 */
+	public static Condition<Bundle> sameAs(Bundle bundle) {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java:57]: I think `sameAs` is the wrong name. In assertj, sameAs is used to mean object identity (`==`) not equality (`Object.equals(Object)`) which is what you are using in all your `sameAs` method. These methods should be called `isEqualsTo` to `equalsTo`.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+	 * 				// ObjectAssert}
+	 * 		.is(sameAs(serviceReference));// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param serviceReferences - the expected serviceReferences that would be
+	 *            checked against other {@link ServiceReference}s
+	 * @return the Condition<br>
+	 */
+	static Condition<ServiceReference<?>> sameAs(ServiceReference<?> serviceReference) {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java:78]: Change name from `sameAs`.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java [hunk 17]
```diff
@@ -0,0 +1,335 @@
...
+	 * 		.have(bundleIsNotNull())
+	 * 		.filteredOn(bundleIsNotNull())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(bundleIsNotNull());// used on {@link ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	public static Condition<FrameworkEvent> bundleIsNotNull() {
```
Comment (bjhargrave) [inline org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java:94]: Do we need to make both `is` and `isNot` versions of different conditions? People can use the `Assertions.not` method to do `not(bundleIsNull())`. This also reduces the API size and testing complexity.

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java (hunk 12 above)
Comment (bjhargrave) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java:169]: OK, but then the javadoc comment above is not commenting on an extant member.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

**Comment by stbischof** [comment]
will add changes in next days

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java (hunk 16 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java]: use: isEqualTo

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java (hunk 15 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundle/BundleConditions.java]: use: isEqualTo

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java (hunk 14 above)
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java [hunk 18]
```diff
@@ -0,0 +1,201 @@
...
+import org.osgi.framework.Constants;
+import org.osgi.framework.Filter;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+/**
+ * A Utility-Class thats Provides static methods to create {@link Condition}s
+ * for an {@link ServiceReference}
```
[the remark this answers is not in the document]
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServiceReferenceConditions.java:39]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServicePropertiesConditions.java [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+
+import org.assertj.core.api.Condition;
+import org.osgi.framework.Filter;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+/**
+ * A Utility-Class thats Provides public static methods to create
+ * {@link Condition}s for ServiceProperties
```
[the remark this answers is not in the document]
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/ServicePropertiesConditions.java:36]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java [hunk 20]
```diff
@@ -0,0 +1,327 @@
...
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.assertj.servicereference.ServicePropertiesConditions;
+import org.osgi.test.assertj.servicereference.ServiceReferenceConditions;
+import org.osgi.test.common.bitmaps.Bitmap;
+import org.osgi.test.common.bitmaps.ServiceEventType;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+/**
+ * A Utility-Class thats Provides public public static methods to create {@link Condition}s
+ * for an {@link ServiceEvent}
```
[the remark this answers is not in the document]
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/serviceevent/ServiceEventConditions.java]: fixed

File: org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java [hunk 21]
```diff
@@ -0,0 +1,296 @@
...
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleEvent;
+import org.osgi.test.assertj.bundle.BundleConditions;
+import org.osgi.test.common.bitmaps.Bitmap;
+import org.osgi.test.common.bitmaps.BundleEventType;
+
+/**
+ * A Utility-Class thats Provides public static methods to create
+ * {@link Condition}s for an {@link BundleEvent}
+ */
```
[the remark this answers is not in the document]
Comment (stbischof) [reply org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventConditions.java:38]: fixed

**Comment by stbischof** [comment]
@bjhargrave could you please review again?

**Pull request closed without merging**